src/docfx/localization/LocalizationConvention.cs

```diff
@@ -122,6 +120,52 @@ public static bool TryGetContributionBranch(string branch, out string contributi
             return false;
         }
 
+        public static (Error error, string content, Document file) TryResolveFromHistory(Docset docset, string pathToDocset)
+        {
+            // try to resolve from source repo's git history
+            // todo: support code snippet
+            var fallbackDocset = GetFallbackDocset();
+            if (fallbackDocset != null && Document.GetContentType(pathToDocset) == ContentType.Page)
+            {
+                var (repo, pathToRepo) = fallbackDocset.RepositoryProvider.GetRepository(fallbackDocset, pathToDocset);
+                if (repo != null)
+                {
+                    var repoPath = PathUtility.NormalizeFolder(repo.Path);
+                    var gitCommitProvider = fallbackDocset.RepositoryProvider.GetGitCommitProvider(repo.Path);
+                    var commits = gitCommitProvider.GetCommitHistory(pathToRepo);
+                    if (commits.Count > 1)
+                    {
+                        // the latest commit would be deleting it from repo
+                        if (GitUtility.TryGetContentFromHistory(repoPath, pathToRepo, commits[1].Sha, out var content))
+                        {
+                            var (error, doc) = Document.TryCreate(fallbackDocset, pathToDocset, isFromHistory: true);
+                            return (error, content, doc);
+                        }
+                    }
+                }
+            }
+
+            return default;
+
+            Docset GetFallbackDocset()
+            {
+                if (docset.LocalizationDocset != null)
+                {
+                    // source docset in loc build
+                    return docset;
```

Comment:
so English also fallback to git history if missing? It shouldn't, add a test for that.